nx-X11/programs/Xserver/dix/resource.c

```diff
@@ -184,6 +184,8 @@ CreateNewResourceType(DeleteType deleteFunc)
        Atom *newnames;
        newnames = realloc(ResourceNames, (next + 1) * sizeof(Atom));
        if(!newnames)
+           free(funcs);
+           free(newnames);
```

Comment:
no need to free newnames since the malloc failed.

Reply:
removed